src/cdk/overlay/scroll/close-scroll-strategy.ts

```diff
@@ -31,19 +41,27 @@ export class CloseScrollStrategy implements ScrollStrategy {
     this._overlayRef = overlayRef;
   }
 
-  /** Enables the closing of the attached on scroll. */
+  /** Enables the closing of the attached overlay on scroll. */
   enable() {
-    if (!this._scrollSubscription) {
-      this._scrollSubscription = this._scrollDispatcher.scrolled(0).subscribe(() => {
-        this._ngZone.run(() => {
-          this.disable();
-
-          if (this._overlayRef.hasAttached()) {
-            this._overlayRef.detach();
-          }
-        });
-      });
+    if (this._scrollSubscription) {
+      return;
     }
+
+    const threshold = this._config ? Math.max(1, this._config.threshold || 0) : 1;
+    const stream = this._scrollDispatcher.scrolled(0);
+    const detach = () => {
+      this._ngZone.run(() => {
+        this.disable();
+
+        if (this._overlayRef.hasAttached()) {
+          this._overlayRef.detach();
+        }
+      });
+    };
+
+    this._scrollSubscription = threshold > 1 ? stream.pipe(take(threshold)).subscribe(() => {
```

Comment:
I don't follow why this passes `threshold` to `take`

Reply:
When we pass it to `take`, it'll call the subscription callback to update the position, until it hits the threshold when it'll complete the observable which detaches the overlay.

Reply:
Isn't `take` the number of event, though? I don't see how it ties to number of pixels

Reply:
The `scrollDispatcher.scrolled` fires once per pixel scrolled, unless it's debounced but we've disabled it by passing in zero.

Reply:
I'd be skeptical that that would be true across all platforms and screen-sizes. Either way, though, a user could also scroll back and forth a bit, no?

Reply:
I'm pretty sure that even mobile devices fire for every pixel now, but you're right that people could scroll back and forth. What do you think about saving the scroll position on open and then checking that you've scrolled X amount of pixels above/below it?

Reply:
That seems reliable